File: C#/InterviewCampProblems/InterviewCampProblems.Code/Class3.cs
/*
Given a sentence, reverse the words of the sentence.For example,"i live in a house" becomes "house a in live i"
*/
using System.Text;

namespace InterviewCampProblems.Code {
    public class Class3 {

        public static string ReverseSentence (string sentence) {

            if (string.IsNullOrEmpty (sentence)) {
                System.Console.WriteLine ("Null or empty string");
                return sentence;
            }
            StringBuilder stringBuilder = new StringBuilder ();
            char[] arr = sentence.ToCharArray ();
            int length = 0;

            for (int i = arr.Length - 1; i >= 0; i--) {

                if (i == 0) {
                    stringBuilder.Append (sentence.Substring (i, length + 1));
                } else if (arr[i] == ' ') {
                    stringBuilder.Append (sentence.Substring (i + 1, length) + ' ');
                    length = 0;
                } else {
                    length++;
                }
            }
            return stringBuilder.ToString ();
        }
    }
}

File: Java/InterviewCamp/src/main/java/Class3.java
//Given a sentence, reverse the words of the sentence.For example,"i live in a house" becomes "house a in live i"

public class Class3 {

    public static String ReverseWordsOfStrings(String str)
    {
        if(str == null || str.isEmpty())
        {
            return  null;
        }

        int i = str.length()-1, end = str.length();
        StringBuilder sb = new StringBuilder();

        while(i >0)
        {
            if(str.charAt(i) == ' ')
            {
                if(sb.length() > 0)
                {
                    sb.append(' ');
                }

                sb.append(str.substring(i+1,end));

                end = i;
            }

            i--;
        }

            if(sb.length() > 0)
            {
                sb.append(' ');
            }

            sb.append(str.substring(i,end));


            return sb.toString();
    }
}
